### PR TITLE
[telemetry_autoscaling] Add telemetry_autoscaling role and zuul job

### PR DESCRIPTION
**Pull request by elfiesmelfie**
The role is imported from the infrawatch/telemetry_autoscaling repo. Moving it here will let us take advantage of zuul testing

### CODE REVIEW

**Comment by elfiesmelfie** [comment]
recheck

**Comment by mgirgisf** [comment]
recheck

File: ci/autoscaling_osp18.yaml
Comment (mgirgisf) [inline ci/autoscaling_osp18.yaml:1]: This file is no longer used

File: ci/functional_tests.yml
Comment (mgirgisf) [inline ci/functional_tests.yml:1]: This file for upcoming functional tests integration.
we can remove it for now also.

File: ci/vars-functional-test.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+# temp: skip os-net-setup
+cifmw_os_net_setup_config: []
```
Comment (mgirgisf) [inline ci/vars-functional-test.yml:3]: I think skip os-net is no longer needed as we use post_deploy hook

**Review by mgirgisf — COMMENTED** [review]
We have a lot of refactor for the code itself, but at least for now we have the main job with autoscaling test

File: ci/patch.yml
Comment (elfiesmelfie) [inline ci/patch.yml:1]: This should be renamed to openstackversion_autoscaling.yaml, or something similar

File: ci/telemetry_autoscaling.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            name: telemetry_autoscaling
+            tasks_from: creating_stack
+          tags:
+            - create
+
+        - import_role:
+            name: telemetry_autoscaling
+            tasks_from: test_autoscaling
+          tags:
+            - test
```
Comment (elfiesmelfie) [inline ci/telemetry_autoscaling.yml:50]: This should be replaced with just the one import_role task.

When we add the logging tests, we can have a list of functional_tests (defaults to a list for all roles to run for OSP18).
Each of the test roles we want to run will be included here, and each will be guarded with `when: "test_autoscaling" in functional_tests`

```
- import_role:
     name: telemetry_autoscaling
   when: '"telemetry_autoscaling" in functional_tests'

- import_role:
      name: test_logging
   when: '"test_logging" in functional_tests' 
 ```

Or we can use a with_items: <functional_tests>, and import_role: item
e.g. 
```
- import_role:
    name: "{{ item }}"
  with_items: "{{ functional_tests }}"
```

Comment (mgirgisf) [reply ci/telemetry_autoscaling.yml:50]: I was thinking to have a separate playbook for each test , and just call it from the run_playbooks but the idea to have them as a list is much better in scaling and we can rename the telemetry_autoscaling.yml to something more generic to include all the telemetry_tests with osp18

**Review by mgirgisf — APPROVED** [review]
the roles need some refactoring and also the vars but we need it to be merged to start refactoring. I approve it for now as it will not cause any conflict with the running job for osp17.

File: ci/vars-functional-test.yml (hunk 1 above)
Comment (elfiesmelfie) [reply ci/vars-functional-test.yml:3]: I want to test that separately, just to make sure.

**Pull request merged**